src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -567,4 +571,64 @@ public virtual void PrintMessageNormalized(TextWriter writer, string message, bo
         else if (!removeLastNewLine)
             writer.WriteLine();
     }
+
+    /// <summary>
+    /// Adds a new runtime option. Throws if option already exists.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    public void AddOption<T>(string name, T value)
+    {
+        Options.Add(name, value);
```

Comment:
>Options.Add(name, value); [](http://example.com/codeflow?start=8&length=25)

if the name added before this will throw, is this the behavior we need? 

The other option can have SetOption instead of AddOption and then allow it overwrite the previous value. 

`Options[name] = value;`

Reply:
So basically `TryAddOption` and `SetOption`? Sounds good to me.

Reply:
Fixed.